### PR TITLE
maui-shell: init at 0.5.6

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Continuation of https://github.com/NixOS/nixpkgs/pull/169055

Run to test
```
nix build -L -f . maui-shell
./result/bin/cask
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/cask-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, pkgs
+, mkDerivation
+, fetchFromGitHub
+, cmake
+, extra-cmake-modules }:
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/cask-server/default.nix:6]:
```suggestion
, extra-cmake-modules
}:
```

File: pkgs/desktops/maui/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  extra-cmake-modules, kwindowsystem, kconfig, knotifications, ki18n, kio,
+  krunner, kactivities, kwayland, prison, kpackage, knotifyconfig, kidletime,
+  kpeople, kdesu, kactivities-stats,
+  ktexteditor, kinit, kunitconversion,
+  kitemmodels, phonon, polkit-qt, kauth,
+  polkit-kde-agent, polkit, mauikit, mauikit-filebrowsing,
+  bluedevil, plasma-pa, plasma-nm, kdelibs4support, bluez-qt
+, maui-core
+, cask-server
+, mauiman }:
```
Comment (SuperSandro2000) [inline pkgs/desktops/maui/default.nix:14]:
```suggestion
, extra-cmake-modules, kwindowsystem, kconfig, knotifications, ki18n, kio
, krunner, kactivities, kwayland, prison, kpackage, knotifyconfig, kidletime
, kpeople, kdesu, kactivities-stats
, ktexteditor, kinit, kunitconversion
, kitemmodels, phonon, polkit-qt, kauth
, polkit-kde-agent, polkit, mauikit, mauikit-filebrowsing
, bluedevil, plasma-pa, plasma-nm, kdelibs4support, bluez-qt
, maui-core
, cask-server
, mauiman
}:
```

File: pkgs/desktops/maui/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      krunner kactivities qtbase qtquickcontrols2 
+      kwayland prison kpackage knotifyconfig kidletime 
+      kpeople kdesu kactivities-stats
+      ktexteditor kinit kunitconversion
+      kitemmodels phonon polkit-qt kauth
+      polkit-kde-agent polkit mauikit mauikit-filebrowsing
+      plasma-nm bluedevil kdelibs4support plasma-pa bluez-qt
+      maui-core
+      mauiman
+      cask-server
```
Comment (SuperSandro2000) [inline pkgs/desktops/maui/default.nix:40]:
```suggestion
  buildInputs = [
    extra-cmake-modules kwindowsystem 
    kconfig knotifications ki18n kio 
    krunner kactivities qtbase qtquickcontrols2 
    kwayland prison kpackage knotifyconfig kidletime 
    kpeople kdesu kactivities-stats
    ktexteditor kinit kunitconversion
    kitemmodels phonon polkit-qt kauth
    polkit-kde-agent polkit mauikit mauikit-filebrowsing
    plasma-nm bluedevil kdelibs4support plasma-pa bluez-qt
    maui-core
    mauiman
    cask-server
```

File: pkgs/desktops/maui/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  enableParallelBuilding = true;
+
```
Comment (SuperSandro2000) [inline pkgs/desktops/maui/default.nix:44]:
```suggestion
```
set by cmake

File: pkgs/development/libraries/maui-core/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, pkgs
+, mkDerivation
+, libcanberra
+, pulseaudio
+, fetchFromGitHub
+, cmake
+, extra-cmake-modules
+, kio
+, kidletime }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/maui-core/default.nix:10]:
```suggestion
, kidletime
}:
```

File: pkgs/development/libraries/maui-core/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+
+  meta = with lib; {
+    description = "Core libraries to manage the desktop to be shared between Maui Settings and Cask";
+    homepage = "https://github.com/Nitrux/maui-core";
+    license = licenses.lgpl3;
+    maintainers = with maintainers; [ onny ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/maui-core/default.nix:42]:
```suggestion
  };
}
```

**Comment by raymens** [comment]
Nice! I tried some time ago but couldn't get it ready.

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -6000,6 +6000,8 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
   };
 
+  cask-server = libsForQt5.callPackage ../applications/misc/cask-server { };
```
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix:6003]: Shouldn't this be in `qt5-packages.nix`?

**Pull request merged**